### ISSUE TITLE
Linux version 4.13.6+ 17.10 (artful)  FuryX

### ISSUE THREAD

**Issue by DarkFoss** [M-Bab/linux-kernel-amdgpu-binaries#34]
Ahh the joys of code churn New drm related message making this more for a reference point for future kernels. I'm sure it will get fixed as those different branches continue to get merged
```
Oct 13 11:23:37 Tardis-1 kernel: [    2.006875] [drm] dc_link_detect: mode number = 0, format_code = 1, channel_count = 1, sample_rate = 7, sample_size = 7
Oct 13 11:23:37 Tardis-1 kernel: [    2.007275] ------------[ cut here ]------------
Oct 13 11:23:37 Tardis-1 kernel: [    2.007283] WARNING: CPU: 0 PID: 119 at drivers/gpu/drm/drm_mode_object.c:237 drm_object_property_set_value+0x57/0x70 [drm]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007283] Modules linked in: amdkfd amd_iommu_v2 crct10dif_pclmul crc32_pclmul ghash_clmulni_intel amdgpu(+) pcbc mxm_wmi aesni_intel aes_x86_64 chash crypto_simd i2c_algo_bit glue_helper ttm cryptd drm_kms_helper syscopyarea sysfillrect sysimgblt fb_sys_fops e1000e drm ahci ptp libahci pps_core wmi
Oct 13 11:23:37 Tardis-1 kernel: [    2.007293] CPU: 0 PID: 119 Comm: kworker/0:2 Not tainted 4.13.6+ #2
Oct 13 11:23:37 Tardis-1 kernel: [    2.007294] Hardware name: To be filled by O.E.M. To be filled by O.E.M./CROSSHAIR V FORMULA-Z, BIOS 2201 03/23/2015
Oct 13 11:23:37 Tardis-1 kernel: [    2.007297] Workqueue: events work_for_cpu_fn
Oct 13 11:23:37 Tardis-1 kernel: [    2.007298] task: ffff9947cd118000 task.stack: ffffa53b81e04000
Oct 13 11:23:37 Tardis-1 kernel: [    2.007303] RIP: 0010:drm_object_property_set_value+0x57/0x70 [drm]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007304] RSP: 0018:ffffa53b81e07b88 EFLAGS: 00010246
Oct 13 11:23:37 Tardis-1 kernel: [    2.007305] RAX: ffffffffc0737bc0 RBX: ffff9947c1360000 RCX: ffff9947c13601a0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007306] RDX: 0000000000000000 RSI: ffff9947c213b780 RDI: ffff9947c1360028
Oct 13 11:23:37 Tardis-1 kernel: [    2.007306] RBP: ffffa53b81e07b88 R08: ffff9947c1360148 R09: ffffffffc06a0cd2
Oct 13 11:23:37 Tardis-1 kernel: [    2.007307] R10: ffffce321103f400 R11: 0000000000000001 R12: 0000000000000000
Oct 13 11:23:37 Tardis-1 kernel: [    2.007307] R13: ffff9947c1330000 R14: ffff9947c1360028 R15: ffff9947c14ed800
Oct 13 11:23:37 Tardis-1 kernel: [    2.007308] FS:  0000000000000000(0000) GS:ffff9947dfc00000(0000) knlGS:0000000000000000
Oct 13 11:23:37 Tardis-1 kernel: [    2.007309] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Oct 13 11:23:37 Tardis-1 kernel: [    2.007310] CR2: 000055c6a6f45fe0 CR3: 0000000331a09000 CR4: 00000000000406f0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007310] Call Trace:
Oct 13 11:23:37 Tardis-1 kernel: [    2.007375]  amdgpu_dm_add_sink_to_freesync_module+0x8d/0x1b0 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007419]  amdgpu_dm_update_connector_after_detect+0xc9/0x240 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007462]  dm_hw_init+0x9d6/0xfb0 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007497]  amdgpu_device_init+0x106f/0x1740 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007499]  ? kmalloc_order+0x18/0x40
Oct 13 11:23:37 Tardis-1 kernel: [    2.007500]  ? kmalloc_order_trace+0x24/0xa0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007534]  amdgpu_driver_load_kms+0x8b/0x2e0 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007539]  drm_dev_register+0x146/0x1d0 [drm]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007572]  amdgpu_pci_probe+0x118/0x140 [amdgpu]
Oct 13 11:23:37 Tardis-1 kernel: [    2.007574]  local_pci_probe+0x45/0xa0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007576]  work_for_cpu_fn+0x14/0x20
Oct 13 11:23:37 Tardis-1 kernel: [    2.007577]  process_one_work+0x1e9/0x3f0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007578]  worker_thread+0x22b/0x410
Oct 13 11:23:37 Tardis-1 kernel: [    2.007579]  kthread+0x109/0x140
Oct 13 11:23:37 Tardis-1 kernel: [    2.007580]  ? process_one_work+0x3f0/0x3f0
Oct 13 11:23:37 Tardis-1 kernel: [    2.007581]  ? kthread_create_on_node+0x70/0x70
Oct 13 11:23:37 Tardis-1 kernel: [    2.007582]  ? kthread_create_on_node+0x70/0x70
Oct 13 11:23:37 Tardis-1 kernel: [    2.007583]  ret_from_fork+0x25/0x30
Oct 13 11:23:37 Tardis-1 kernel: [    2.007584] Code: 38 85 ff 7e 24 49 3b 70 08 74 24 49 8d 48 10 31 c0 eb 0a 48 83 c1 08 48 3b 71 f8 74 14 83 c0 01 39 f8 75 ef b8 ea ff ff ff 5d c3 <0f> ff eb c6 31 c0 48 98 49 89 94 c0 c8 00 00 00 31 c0 5d c3 0f 
Oct 13 11:23:37 Tardis-1 kernel: [    2.007601] ---[ end trace 946ccb2716487e7b ]---
Oct 13 11:23:37 Tardis-1 kernel: [    2.008029] [drm] Supports vblank timestamp caching Rev 2 (21.10.2013).

```
also lead to a lot of TTM (over a hundred)spam may have triggered a bug in evolution:
```
Oct 13 11:24:12 Tardis-1 evolution-calen[1837]: e_cal_recur_generate_instances_sync: assertion 'icaltime_compare (interval_start, interval_end) < 0' failed
Oct 13 11:24:22 Tardis-1 kernel: [   62.615614] [TTM] Erroneous page count. Leaking pages.
Oct 13 11:24:14 Tardis-1 evolution-calen[1837]: message repeated 141 times: [ e_cal_recur_generate_instances_sync: assertion 'icaltime_compare (interval_start, interval_end) < 0' failed
```]
It did boot  up fine however I just skipped doing any testing too spammy for my taste sticking with 4.13.5+ for now :)

**Comment by DarkFoss** [comment]
Realized I've never left any specs system at a glance:
```
OS: Ubuntu 17.10 artful
Kernel: x86_64 Linux 4.13.0-15-generic
Uptime: 1d 5h 28m
Packages: 1983
Shell: bash 4.4.12
DE: GNOME
WM: GNOME Shell
WM Theme: Adwaita
GTK Theme: ubuntu-mono-dark
Icon Theme: Adwaita
Font: Ubuntu 11
CPU: AMD FX-8350 Eight-Core @ 8x 4GHz [27.0°C]
MB: CROSSHAIR V FORMULA-Z, BIOS 2201 03/23/2015
RAM:Crucial Ballistix Elite 16GB DDR3@1866 (2x8gb) ===|===1598MiB / 16021MiB
GPU: AMD Radeon (TM) R9 Fury Series (AMD FIJI / DRM 3.18.0 / 4.13.0-12-generic, LLVM 5.0.0)
Monitor: Samsung S23A950D (16:9) 119.98 Hz displayport
Resolution: 1920x1080
```

**Comment by piet8stevens** [comment]
Does sound over hdmi work for you on 17.10 with this kernel? I have it working on 17.04 (with 4.12.10+ kernel) but hesitate to upgrade until I know it is OK.

**Comment by daniel-jirca** [comment]
@piet8stevens You can always revert to the previous kernel from the grub menu if audio doesn't work for you. Go ahead and try it. It works on Mint 18.2 (ubuntu 16.04.3).

**Comment by M-Bab** [comment]
@piet8stevens I am running Xubuntu 17.10 with my kernel since its release without issues.

**Comment by M-Bab** [comment]
Can you recheck with 4.17?

**Comment by DarkFoss** [comment]
Sorry M-BAB,
I had a partially broken install from switching between oibaf's and padoka's ppa's. I took a break and did some Win gaming. I got motivated to fix and upgrade to 18.04 when Steam Play came out.

All of those old issues are gone with your 18.5 Ubuntu kernel runs well ! A few dmesg errors I'll uninstall the Ubuntu flavor and try the stock staging before posting a new thread. I don't use hdmi for sound can't comment on @piet8stevens  issue you can close this if you'd like
